h/search/query.py

```diff
@@ -142,10 +142,6 @@ def _parse_date(str_value):
         except ValueError:
             try:  # pylint: disable=too-many-try-statements
                 date = parse(str_value, default=DEFAULT_DATE)
-                # If timezone isn't specified assume it's utc.
-                if not date.tzinfo:
-                    date = date.replace(tzinfo=tz.tzutc())
-
```

Comment:
There's a comment in the commit about this, but the DEFAULT_DATE provides values for every missing component, including TZ.